### ISSUE TITLE
Investigate GraphQL

### ISSUE THREAD

**Issue by 0xazure** [0xazure/supernova#31]
Currently we're using GitHub's API `v3` that returns JSON data. GitHub also has a `v4` endpoint that [supported GraphQL](https://developer.github.com/v4/) which might be of interest since we do not use most of the JSON response data we receive from the `v3` endpoint, and GraphQL will let us better tailor our requests to GitHub's API.

We will need to investigate what options are available for HTTP request libraries that can work with GraphQL. [This search](https://crates.io/search?q=graphql) on crates.io may be a good start.

**Comment by biskit1** [comment]
can i try looking into this?

**Comment by seanprashad** [comment]
It might be best to finish off your existing PRs (#34 and #39) first! 👊🏽

**Comment by biskit1** [comment]
Looks like this issue would involve refactoring quite a bit of our code, and would require more learning time than I have available at the moment. @0xazure if you want help with this in the future and can provide more specific/detailed direction, I'd be happy to contribute.

**Comment by 0xazure** [comment]
Yeah this definitely isn't a small piece of work, we'd be changing pretty much everything about the way we interact with the GitHub API.

Part of this issue is going to be specifying what exactly we need to research, with a focus on if migrating to GraphQL is worth the investment.

Some things I can think of, off the top of my head:

1. what option are there for GraphQL libraries in Rust?
1. will using GraphQL save the amount of data sent over the wire (we don't currently deserialize everything GitHub sends us, and this can be kind of wasteful)
1. does using the GraphQL endpoint eliminate the need for pagination (this is a big one, and would simplify a lot of our requests code)
1. how does the GraphQL endpoint implement request limits, if at all?
